src/platform/packages/shared/shared-ux/button_toolbar/src/buttons/icon_button_group/icon_button_group.styles.ts

```diff
@@ -35,6 +35,11 @@ export const IconButtonGroupStyles = ({ euiTheme }: UseEuiTheme) => {
     buttonGroup: {
       '.euiButtonGroup__buttons': {
         borderRadius: `${euiTheme.border.radius.medium}`,
+        '.nextToSearchInput': {
+          borderLeft: 'none !important',
+          borderTopLeftRadius: `0px !important`,
+          borderBottomLeftRadius: `0px !important`,
+        },
```

Comment:
I guess this is a good example of why the search input could be a component that encapsulates both the button and the search input. Similar to the [`<InTableSearchControl />`](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-data-grid-in-table-search/src/in_table_search_control.tsx#L170). This way, the component itself is responsible for its style.

My main concern here is that this `.nextToSearchInput` css class is not part of `<IconButtonGroup />` component, but something we are customizing on the metrics grid.

Reply:
I refactored it so we use emotion CSS and make the component styles customizable - then passed our overrides there - that way the component can be used with different styling + our overrides live in our component, wdyt?